lib/flutter_webview_plugin.dart

```diff
@@ -1,33 +1,41 @@
 import 'dart:async';
 
 import 'package:flutter/services.dart';
+import 'package:flutter/material.dart';
 
 const _kChannel = 'flutter_webview_plugin';
+const _kEvent = 'flutter_webview_plugin_event';
 
 /// Singleton Class that communicate with a fullscreen Webview Instance
 /// Have to be instanciate after `runApp` called.
 class FlutterWebviewPlugin {
-  final MethodChannel _channel = const MethodChannel(_kChannel);
+  final MethodChannel _channel;
+
+  final EventChannel _event;
+  Stream<String> _stateChanged;
+
+  Stream<String> get stateChanged {
+    if (_stateChanged == null) {
+      _stateChanged = _event.receiveBroadcastStream();
+    }
+    return _stateChanged;
+  }
+
   final StreamController<Null> _onDestroy = new StreamController.broadcast();
   final StreamController<Null> _onBackPressed =
       new StreamController.broadcast();
 
-  /// TODO: iOS implementation
   final StreamController<String> _onUrlChanged =
       new StreamController.broadcast();
 
-  static FlutterWebviewPlugin _instance;
-  FlutterWebviewPlugin._() {
-    _init();
-  }
-
-  factory FlutterWebviewPlugin() => _instance ??= new FlutterWebviewPlugin._();
-
-  _init() {
+  FlutterWebviewPlugin() : 
+    _channel = const MethodChannel(_kChannel),
+    _event = const EventChannel(_kEvent) {
```

Comment:
why a new channel and not use the existing MethodChannel ?

Reply:
I don't know how to send event via MethodChannel. 
https://github.com/flutter/plugins/blob/master/packages/battery/lib/battery.dart